src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -32,6 +35,8 @@ namespace Microsoft.CodeAnalysis
     /// </remarks>
     public abstract class GeneratorDriver
     {
+        private static readonly ConditionalWeakTable<SourceText, SyntaxTree> s_parsedGeneratedSources = new();
```

Comment:
Consider that recently we evolved the `AnalyzerDriver` from a model where we implemented caching via `ConditionalWeakTable` to an explicit provider model. Why are we choosing to use `ConditionalWeakTable` here instead of an explicit provider? The latter seems to be the direction we want to go to give more control to the host (IDE or compiler) on how caching should be implemented.

Reply:
I wasn't aware of the other work (for reference it was completed in #46508). I used a `SyntaxTreeProvider`, but it's not clear to me how the IDE would use a different path from compilation considering these are internal APIs.